pynetgear/__main__.py

```diff
@@ -1,22 +1,101 @@
 """Run PyNetgear from the command-line."""
 import sys
+import os
+
+from argparse import ArgumentParser
 from pynetgear import Netgear
 
 
+def make_formatter(format_name):
+    """Returns a callable that outputs the data. Defaults to csv."""
+
+    if format_name == "json":
+        from json import dumps
+
+        def jsonify(devices):
+            print(dumps([device._asdict() for device in devices]))
+        return jsonify
+    else:
+        return (lambda devices: [print(device) for device in devices])
+
+
+def argparser():
+    parser = ArgumentParser(prog='pynetgear')
+
+    parser.add_argument("--format", choices=['json', 'py'], default='py')
+    # parser.add_argument("--router-config", type=FileType('r'), help="Read router config from file.")
```

Comment:
I wanted to add a read from file instead of having to specify things at the CLI too often, but that seems premature and then handling conflicts between the two.